provider/apply.go

```diff
@@ -270,6 +273,46 @@ func (s *RawProviderServer) ApplyResourceChange(ctx context.Context, req *tfprot
 			return resp, nil
 		}
 
+		// wait for the resource to be deleted
+		s.logger.Trace("[ApplyResourceChange][Apply]", "Waiting for resource to be deleted...")
+		timeout := defaultDeleteTimeout
+		if v, ok := priorStateVal["timeouts"]; ok && !v.IsNull() {
+			timeouts := map[string]tftypes.Value{}
+			v.As(&timeouts)
+			if v, ok := timeouts["delete"]; ok && !v.IsNull() {
+				v.As(&timeout)
+			}
+		}
+		s.logger.Trace("[ApplyResourceChange][Apply]", "Using timeout %q for delete", timeout)
+		duration, err := time.ParseDuration(timeout)
+		if err != nil {
```

Comment:
I feel like it would be a better user experience to detect syntactic errors (as returned by `ParseDuration`) earlier, during `ValidateResourceConfig` or during `PlanResourceChange`. Failing the operation during these phases is less disruptive to thee user. (Also, a successful plan should not be followed by an `apply` failing because of wrong syntax in config).

Reply:
Have added some validation code so the user gets this if the format is incorrect:

```
╷
│ Error: Failed to parse timeout for "delete"
│
│   with kubernetes_manifest.test,
│   on main.tf line 45, in resource "kubernetes_manifest" "test":
│   45:     delete = "sausages"
│
│ time: invalid duration "sausages"
╵
```